spec/anumerable_spec.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+require './methods.rb'
+describe Enumerable do
+  let(:test_arr) { [1, 2, 3] }
+  let(:test_block) { proc { |i| i * 2 } }
+  let(:test_arr_str) { %w[one two three] }
+  let(:test_length) { proc { |word| word.length >= 3 } }
+  let(:test_length_2) { proc { |word| word.length >= 5 } }
+  let(:test_range) { (1..4) }
+  let(:test_absolutes) { [nil, true, false] }
+  let(:test_numeric) { [1, 2i, 3.14] }
+
+  describe '#my_each' do
+    it 'returns original array when no output' do
+      expect(test_arr.my_each { test_block }).to eq(test_arr)
+    end
+
+    it 'prints result to stdout from block calculation i * 2' do
+      expect { test_arr.my_each { |i| print i * 2 } }.to output(/246/).to_stdout
+    end
+
+    it 'returns an Enumerator when no block given' do
+      expect(test_arr.my_each.is_a?(Enumerator)).to be(true)
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'It expect to be equal to 1 2 3' do
+      expect(test_arr.my_each_with_index { test_block }).to eql(test_arr)
+    end
+
+    it 'returns an Enumerator when no block given' do
+      expect(test_arr.my_each_with_index.is_a?(Enumerator)).to eql(true)
+    end
+  end
+
+  describe '#my_select' do
+    it 'Check if its an even' do
+      expect(test_arr.my_select(&:even?)).to eql([2])
+    end
+
+    it 'return an Enumerable when no block given' do
+      expect(test_arr.my_select.is_a?(Enumerable)).to eql(true)
+    end
+  end
+
+  describe '#my_all?' do
+    it 'Check the length of the words 1 with 3' do
+      expect(test_arr_str.my_all?(&test_length)).to eql(true)
+    end
+
+    it 'Check the length of the words 2 with 5' do
+      expect(test_arr_str.my_all?(&test_length_2)).to eql(false)
+    end
+  end
```

Comment:
By adding this test, 
```ruby
it 'return false when no block is given' do
      expect([1, false, 'hi', []].my_all?).to eql([1, false, 'hi', []].all?)
    end
```
I'm getting that error
```ruby
Failures:

  1) Enumerable#my_all? return false when no block is given
     Failure/Error: expect([1, false, 'hi', []].my_all?).to eql([1, false, 'hi', []].all?)
     
       expected: false
            got: true
     
       (compared using eql?)
     
       Diff:
       @@ -1,2 +1,2 @@
       -false
       +true
       
     # ./spec/anumerable_spec.rb:58:in `block (3 levels) in <top (required)>'
```

Reply:
Solved, was a little bug with my enumerable code.